tests/test_georaster.py

```diff
@@ -544,33 +544,83 @@ def test_reproject(self, example: str) -> None:
         # Reference raster to be used
         r = gr.Raster(example)
 
-        # Create an artificial raster with different bounds, shape and resolution
-        new_data = np.ones(r.shape)
-        transform = list(r.transform)
-        new_transform = rio.transform.Affine(
-            transform[0] / 2,
-            transform[1],
-            (r.bounds.left + r.bounds.right) / 2,
-            transform[3] / 2,
-            transform[4],
-            (r.bounds.bottom + r.bounds.top) / 2,
+        # -- Check proper errors are raised if nodata are not set -- #
+        r_ndv = r.copy()
+        r_ndv.nodata = None
+        default_ndv = _default_ndv(r_ndv.dtypes[0])
+        r_ndv.data.data[~r_ndv.data.mask][0] = default_ndv
+
+        # 1 - if no src_nodata is set and masked values exist, raises an error
+        if np.sum(r_ndv.data.mask) > 0:
+            with pytest.raises(ValueError, match="No nodata set, use `src_nodata`"):
+                r_ndv.reproject(dst_res=r_ndv.res[0] / 2, dst_nodata=0)
```

Comment:
Do you need to specify `dst_nodata=0` here? Won't the `ValueError` be raised in any case?

Reply:
If `dst_nodata` is not set, it will raise a warning. That's case 2 below.